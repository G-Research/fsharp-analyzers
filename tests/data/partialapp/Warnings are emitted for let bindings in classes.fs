module M

type MyClass() =
    let mutable x = 0
    let myFunc x y = x + y

    let partapp1 =
        myFunc 1 // should warn
        let y = myFunc 23 // should warn
        myFunc 3 44 // should not warn

    let partapp2 = myFunc 4 // should warn
    let partapp3 = (+) 4 55 // should not warn
    let partapp4 = (+) 4 // should warn
    let partapp5: (int seq -> int seq) = Seq.map (fun x -> x + 1) // should warn
