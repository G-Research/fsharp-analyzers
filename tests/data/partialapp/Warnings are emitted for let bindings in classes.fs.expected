GRA-PARTAPP-001 | Warning | (8,8 - 8,14) | partial application should not be used: myFunc at (8,8--8,14)
GRA-PARTAPP-001 | Warning | (9,16 - 9,22) | partial application should not be used: myFunc at (9,16--9,22)
GRA-PARTAPP-001 | Warning | (12,19 - 12,25) | partial application should not be used: myFunc at (12,19--12,25)
GRA-PARTAPP-001 | Warning | (14,20 - 14,21) | partial application should not be used: op_Addition at (14,20--14,21)
GRA-PARTAPP-001 | Warning | (15,41 - 15,48) | partial application should not be used: map at (15,41--15,48)
